e2e/testdata/fn-render/path-index-duplicate/.expected/config.yaml

```diff
@@ -15,4 +15,4 @@
 # One of the functions in the pipeline fails resulting in
 # non-zero exit code and no changes in the resources.
 exitCode: 1
-stdErr: 'resource at path "resources.yaml" and index "0" already exists'
+stdErr: 'resource at path \"resources.yaml\" and index \"0\" already exists'
```

Comment:
```
Because this change is to surface the function results whenever we encounter an error, some output of `kpt fn render` changes.

On main:
```
Package "path-index-duplicate": 
[RUNNING] "gcr.io/kpt-fn/starlark:v0.2.1"
[FAIL] "gcr.io/kpt-fn/starlark:v0.2.1" in 2.2s
Error: resource at path "resources.yaml" and index "0" already exists
```

With this PR:

```
$ kpt fn render e2e/testdata/fn-render/path-index-duplicate
Package "path-index-duplicate": 
[RUNNING] "gcr.io/kpt-fn/starlark:v0.2.1"
[FAIL] "gcr.io/kpt-fn/starlark:v0.2.1" in 1.7s
  Stderr:
    "resource at path \"resources.yaml\" and index \"0\" already exists"
```

If there had been any `Results`, this PR would have surfaced them while `main` would not have. 

I think this is a good change because we should be surfacing stderr in our structured results format wherever possible, as that's what caused the original issue in porch, but I did want to call out that this PR affects both kpt CLI and porch (which is not surprising considering that it is a shared library between them. 
```